app/api/auth/signin.js

```diff
@@ -0,0 +1,26 @@
+import { NextAuth } from "next-auth";
+import { AuthError } from "next-auth";
+
+export default async function handler(req, res) {
+  if (req.method === "POST") {
+    const { provider } = req.body; // Get the provider from the request body
+
+    try {
+      const result = await signIn(provider, { redirect: false });
+
+      if (result.error) {
+        console.error("Sign-in error:", result.error); // Log error to terminal
+        return res.redirect(`/testauth?error=${encodeURIComponent(result.error)}`);
+      }
+
+      // On successful sign-in, redirect to the profile page
+      res.redirect('/profile'); 
```

Comment:
We'll want to redirect to where ever the user came from.
Could you create an issue for this to track for later. It can be a stretch goal.
For now I would suggest:
```suggestion
      res.redirect('/'); 
```